Improve navigation section pages

skyace65 opened the pull request:
Supersedes #6044 and #6053. Per discussions on Godot's rocket.chat the content of those two PRs has been combined to replace the existing 3D introduction page and create a 2D introduction page. @mhilbrunner I know you were waiting on this. I've made some minor tweaks to the content from those PR's and copied over the RVO section from the existing introduction page. @smix8 please look this over as well and see if there's anything that should be changed or fixed, and thank you for the two PR's and other navigation ones, they're a big help.

YuriSizov commented:
I'm assuming that smix8 should be added as a co-author for this one then?

smix8 commented on tutorials/navigation/navigation_introduction_3d.rst:
```diff
@@ -0,0 +1,152 @@
+.. _doc_navigation_overview:
```
`navigation_introduction_3d` or `navigation_overview`, what shall it be? Same for 2D.

Reminder for the need to check the linking of other pages in the open pr's if this gets changed around, can't remember if this page was crosslinked.

smix8 commented on tutorials/navigation/navigation_introduction_3d.rst:
```diff
@@ -0,0 +1,152 @@
+Setup for 3D scene
+------------------
+
+The following steps show how to setup a minimum viable RVO navigation in 3D that uses the NavigationServer3D and
```
This is not correct as the following steps and examples do not include anything about RVO avoidance. The original was "how to setup a minimum viable navigation in 3D".

smix8 commented on tutorials/navigation/navigation_introduction_3d.rst:
```diff
@@ -0,0 +1,152 @@
+    - The NavigationServer3D uses this resource to update navmesh of individual regions.
+    - The GridMap Editor uses this resource when specific navigationmeshes are defined for each gridcell.
+
+RVO processing
```
I am not sure about including so much about the RVO avoidance in was is more or less a navigation beginners page as it is a very overwhelming / confusing topic for new users and there is no steps or examples for it included here. Maybe only mention it as an available feature for the overview with one sentence / little paragraph and a link to the main avoidance page and merge the detail there? Same for 2D.

skyace65 commented:
I've fixed the issues and removed mention of RVO except for the obstacle nodes, this page should be updated in whatever PR adds RVO information so we can link directly to it. Also didn't know co-authoring was a thing, added smix8 as a co-author.

mhilbrunner commented on tutorials/navigation/navigation_introduction_2d.rst:
```diff
@@ -0,0 +1,147 @@
+Introduction
+===================
+
+Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation
```
```suggestion
Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation
```

mhilbrunner commented on tutorials/navigation/navigation_introduction_2d.rst:
```diff
@@ -0,0 +1,147 @@
+
+Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation
+and pathfinding for 2D and 3D games. The following section provides a quick overview over all available
+navigation related objects in Godot for 2D scenes and their primary use.
```
`object`s is used multiple times and strikes me as a bit odd compared to other Godot docs. Shouldn't we call them nodes and classes? Or are these different?

mhilbrunner commented on tutorials/navigation/navigation_introduction_2d.rst:
```diff
@@ -0,0 +1,147 @@
+    The AStar2D class is best suited for cellbased 2D gameplay that does not require actors to reach any possible position within an area but only predefined, distinct positions.
+
+- :ref:`NavigationServer2D<class_NavigationServer2D>`
+    ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a navigationmesh defined area.
```
```suggestion
    ``NavigationServer2D`` provides a powerful server API to find the shortest path between two positions on a area defined by a navigation mesh.
```
`navigationmesh` is used multiple times in the doc, it should IMO either be `navigation mesh`, `navmesh` or link to a class and then be capitalized as `NavigationMesh` if it is one.

mhilbrunner commented on tutorials/navigation/navigation_introduction_3d.rst:
```diff
@@ -0,0 +1,134 @@
+Introduction
+===================
+
+Godot provides multiple objects, classes and servers to facilitate grid-based- or mesh-based navigation
```
```suggestion
Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation
```

mhilbrunner commented on tutorials/navigation/navigation_introduction_3d.rst:
```diff
@@ -0,0 +1,134 @@
+    a Node3D inheriting parent Node.
+
+-  :ref:`NavigationObstacle3D<class_NavigationObstacle3D>` Node
+    A Node that acts as an agent with avoidance radius, to work it just needs to be added under a Node3D
```
```suggestion
    A Node that acts as an agent with avoidance radius, to work it needs to be added under a Node3D
```

mhilbrunner commented on tutorials/navigation/navigation_introduction_2d.rst:
```diff
@@ -0,0 +1,147 @@
+    for a Node2D inheriting parent Node.
+
+-  :ref:`NavigationObstacle2D<class_NavigationObstacle2D>` Node
+    A Node that acts as an agent with avoidance radius, to work it just needs to be added under a Node2D
```
```suggestion
    A Node that acts as an agent with avoidance radius, to work it needs to be added under a Node2D
```

mhilbrunner reviewed (commented):
Thanks to both of you for collaborating on this. Some comments below:

skyace65 commented:
Fixed.

mhilbrunner commented:
Good work, thanks for the collaboration you two, lets get this in and iterate if needed; LGTM.
